### PR TITLE
Updating the react-router version and updating how the main routing base works.

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Version bump and adding a `Main` component so that component routing can work better, and to be easier to remove the `Modal` in the next PR.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -61,8 +61,8 @@ app.use('/', (req, res) => {
   alt.bootstrap(JSON.stringify(res.locals.data || {}));
   const iso = new Iso();
 
-  const blogAppUrl = (req.url).indexOf('books-music-dvds/recommendations/staff-picks') !== -1;
-  const routes = blogAppUrl ? appRoutes.client : appRoutes.server;
+  const serverUrl = (req.url).indexOf('books-music-dvds/recommendations/staff-picks') !== -1;
```
Comment (rhernand3z) [inline server.js]: would be better to use the config.baseUrl value from a previous PR instead? less places to track if the url changes.

Comment (kfriedman) [reply server.js]: FYI, I'm working on updating all the routes to use config.

Comment (EdwinGuzman) [reply server.js]: Yup, I purposely left some for @kfriedman

**Comment by EdwinGuzman** [comment]
Fixed conflicts based on @kfriedman 's PR.

**Pull request merged**